From a493b67f9389df35876e59d4b73bc6af0b746ff9 Mon Sep 17 00:00:00 2001
From: Xenia Ragiadakou <xenia.ragiadakou@amd.com>
Date: Wed, 6 Dec 2023 17:23:31 +0200
Subject: [PATCH 221/221] Revert "x86/cpufeatures: expose TOPOEXT to all
 guests"

This reverts commit db989b6dec36ae4452c07eaea8d64c01eb7407a9.

With this change ROCR fails to retrieve cache info when initializing HSA stack:

:3:rocdevice.cpp            :434 : 0156951423 us: 4707 : [tid:0x7fc84f43d740] Initializing HSA stack.
Fail to get cache info for node 0
Fail to get cache info for node 0
:1:rocdevice.cpp            :442 : 0156952564 us: 4707 : [tid:0x7fc84f43d740] hsa_init failed.
:4:runtime.cpp              :83  : 0156952795 us: 4707 : [tid:0x7fc84f43d740] init

Therefore, revert it until we identify the reason of the failure.

Change-Id: I1fcc942dca0b43dc852a0baa973becc6071c56b1
---
 xen/include/public/arch-x86/cpufeatureset.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/xen/include/public/arch-x86/cpufeatureset.h b/xen/include/public/arch-x86/cpufeatureset.h
index b7de9abf23..7915f5826f 100644
--- a/xen/include/public/arch-x86/cpufeatureset.h
+++ b/xen/include/public/arch-x86/cpufeatureset.h
@@ -165,7 +165,7 @@ XEN_CPUFEATURE(LWP,           3*32+15) /*   Light Weight Profiling */
 XEN_CPUFEATURE(FMA4,          3*32+16) /*A  4 operands MAC instructions */
 XEN_CPUFEATURE(NODEID_MSR,    3*32+19) /*   NodeId MSR */
 XEN_CPUFEATURE(TBM,           3*32+21) /*A  trailing bit manipulations */
-XEN_CPUFEATURE(TOPOEXT,       3*32+22) /*A  topology extensions CPUID leafs */
+XEN_CPUFEATURE(TOPOEXT,       3*32+22) /*   topology extensions CPUID leafs */
 XEN_CPUFEATURE(DBEXT,         3*32+26) /*A  data breakpoint extension */
 XEN_CPUFEATURE(MONITORX,      3*32+29) /*   MONITOR extension (MONITORX/MWAITX) */
 
-- 
2.17.1

